src/test/SecurityPanel/StoreJmxCredentials.test.tsx

```diff
@@ -0,0 +1,256 @@
+
+jest.mock('@app/SecurityPanel/CreateJmxCredentialModal', () => {
+  return {
+    CreateJmxCredentialModal: jest.fn((props) => {
+      return (
+        <Modal
+          isOpen={props.visible}
+          variant={ModalVariant.large}
+          showClose={true}
+          onClose={props.onClose}
+          title="CreateJmxCredentialModal"
+        >
+          Jmx Auth Form
+        </Modal>
+      );
+    }),
+  };
+});
+
+jest.mock('@app/Shared/Services/NotificationChannel.service', () => {
+  const mockNotification = { message: { target: 'service:jmx:rmi://someUrl' } } as NotificationMessage;
+  return {
+    ...jest.requireActual('@app/Shared/Services/NotificationChannel.service'),
+    NotificationChannel: jest.fn(() => {
+      return {
+        messages: jest
+          .fn()
+          .mockReturnValueOnce(of()) // 'renders correctly'
+          .mockReturnValueOnce(of())
+          .mockReturnValueOnce(of())
+
+          .mockReturnValueOnce(of()) // 'adds the correct table entry when a stored notification is received'
+          .mockReturnValueOnce(of())
+          .mockReturnValueOnce(of(mockNotification))
+
+          .mockReturnValueOnce(of()) // 'removes the correct table entry when a deletion notification is received'
+          .mockReturnValueOnce(of(mockNotification))
+          .mockReturnValueOnce(of())
+
+          .mockReturnValueOnce(of()) // 'renders an empty table after receiving deletion notifications for all credentials'
+          .mockReturnValueOnce(of(mockNotification))
+          .mockReturnValueOnce(of())
+
+          .mockReturnValue(of()), // all other tests
+      };
+    }),
+  };
+});
+
+jest.mock('@app/Shared/Services/Target.service', () => {
+  return {
+    ...jest.requireActual('@app/Shared/Services/Target.service'),
+    TargetService: jest.fn(() => {
+      return {
+        deleteCredentials: jest.fn(),
+      };
+    }),
+  };
+});
+
+jest.doMock('@app/Shared/Services/Api.service', () => {
+  return {
+    ...jest.requireActual('@app/Shared/Services/Api.service'),
+    ApiService: jest.fn(() => {
+      return {
+        getTargetsWithStoredJmxCredentials: jest
+          .fn()
+          .mockReturnValueOnce(of([mockTarget]))
+          .mockReturnValueOnce(of([]))
+          .mockReturnValueOnce(of([mockTarget, mockAnotherTarget]))
+          .mockReturnValueOnce(of([mockTarget]))
+          .mockReturnValueOnce(of([]))
+          .mockReturnValueOnce(of([]))
+          .mockReturnValueOnce(of([mockTarget, mockAnotherTarget]))
+          .mockReturnValueOnce(of([mockTarget, mockAnotherTarget])),
+        deleteTargetCredentials: jest.fn(() => {
+          return of(true);
+        }),
+      };
+    }),
+  };
+});
+
+jest.doMock('@app/Shared/Services/Targets.service', () => {
+  return {
+    ...jest.requireActual('@app/Shared/Services/Targets.service'),
+    TargetsService: jest.fn(() => {
+      return {
+        targets: jest.fn().mockReturnValue(of([mockTarget, mockAnotherTarget])),
```

Comment:
This can be changed to a simple `jest.fn()` implementation like `deleteTargetCredentials` above (i.e. no need for `mockReturnValue` since there's no chaining going on), just for consistency reasons.

Reply:
Up to you though, whichever format you prefer.